### ISSUE TITLE
Line of dashes throws "Invalid resource line" in route file

### ISSUE THREAD

**Issue by mrkkrp** [yesodweb/yesod#1180]
Since `--` starts line comments, I think it makes sense to use lines of dashes to divide sections of route files for readability, just how they can be used in Haskell. This however results in:

```
    /home/mark/projects/programs/my-project/Foundation.hs:52:21:
        Exception when trying to run compile-time code:
          Invalid resource line: ----------------------------------------------------------------------------
        Code: parseRoutesFile "config/routes"
        In the splice: $(parseRoutesFile "config/routes")
```

Is this intentional?

**Comment by snoyberg** [comment]
It's the current documented behavior, but I see no harm in adding support
for comment lines. Interested in trying out a PR? Should be relatively
simple.

**Comment by mrkkrp** [comment]
I think I will get to this in a few days.

**Comment by snoyberg** [comment]
Resolved by #1182
